File: BlackBerry/AlertMsgs/PINMsgAlert.java
package com.mlhsoftware.Blurts;

import org.json.me.*;


class PINMsgAlert extends AlertBase 
{
  // Tag Names
  private static final String ALERT_NAME = "PINMsg";
  private static final String KEY_SENDER_NAME = "SenderName";
  private static final String KEY_ACCOUNT = "ReceivingAccount";
  private static final String KEY_ADDRESS = "SenderAddress";
  private static final String KEY_SUBJECT = "Subject";
  private static final String KEY_BODY = "BodyText";


  public PINMsgAlert( String string ) throws JSONException
  {
    super( string );
  }

  public PINMsgAlert()
  {
    super( TYPE_PIN_MSG, ALERT_NAME );
  }

  public void setAccount( String value )
  {
    try
    {
      put( KEY_ACCOUNT, value );
    }
    catch ( JSONException e )
    {
      System.out.println( e.toString() );
    }
  }

  String getAccount()
  {
    return optString( KEY_ACCOUNT );
  }

  public void setName( String value )
  {
    try
    {
      put( KEY_SENDER_NAME, value );
    }
    catch ( JSONException e )
    {
      System.out.println( e.toString() );
    }
  }

  String getName()
  {
    return optString( KEY_ADDRESS );
  }

  public void setAddress( String value )
  {
    try
    {
      put( KEY_ADDRESS, value );
    }
    catch ( JSONException e )
    {
      System.out.println( e.toString() );
    }
  }

  String getAddress()
  {
    return optString( KEY_ADDRESS );
  }

  public void setSubject( String value )
  {
    try
    {
      put( KEY_SUBJECT, value );
    }
    catch ( JSONException e )
    {
      System.out.println( e.toString() );
    }
  }

  String getSubject()
  {
    return optString( KEY_SUBJECT );
  }

  public void setBody( String value )
  {
    try
    {
      put( KEY_BODY, value );
    }
    catch ( JSONException e )
    {
      System.out.println( e.toString() );
    }
  }

  String getBody()
  {
    return optString( KEY_BODY );
  }
}

File: Desktop/Domain/AlertMsgs/PINMsgAlert.cs
using System;
using System.Runtime.InteropServices;
using Blurts.Domain;
namespace Blurts
{
  [ComVisible(true)]
  public class PINMsgAlert : AlertBase
  {
    // Tag Names
    private static String ALERT_NAME = "PINMsg";
    private static String KEY_SENDER_NAME = "SenderName";
    private static String KEY_ACCOUNT = "ReceivingAccount";
    private static String KEY_ADDRESS = "SenderAddress";
    private static String KEY_SUBJECT = "Subject";
    private static String KEY_BODY = "BodyText";


    public PINMsgAlert(String s)
      : base(s)
    {
    }

    public PINMsgAlert()
      : base(DisplayItemType.PIN_MSG, ALERT_NAME)
    {
    }


    [ComVisible(true)]
    public String ReceivingAccount
    {
      get
      {
        return optString(KEY_ACCOUNT);
      }
      set
      {
        try
        {
          remove(KEY_ACCOUNT);
          put(KEY_ACCOUNT, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }

    [ComVisible(true)]
    public String SenderName
    {
      get
      {
        return optString(KEY_SENDER_NAME);
      }
      set
      {
        try
        {
          remove(KEY_SENDER_NAME);
          put(KEY_SENDER_NAME, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }


    [ComVisible(true)]
    public String SenderAddress
    {
      get
      {
        return optString(KEY_ADDRESS);
      }
      set
      {
        try
        {
          remove(KEY_ADDRESS);
          put(KEY_ADDRESS, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }

    [ComVisible(true)]
    public String Subject
    {
      get
      {
        return optString(KEY_SUBJECT);
      }
      set
      {
        try
        {
          remove(KEY_SUBJECT);
          put(KEY_SUBJECT, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }
    

    [ComVisible(true)]
    public String BodyText
    {
      get
      {
        return optString(KEY_BODY);
      }
      set
      {
        try
        {
          remove(KEY_BODY);
          put(KEY_BODY, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }

    public override Boolean ProcessMessage()
    {
      this.ImageFile = Contacts.Instance.getImageFile(null, this.SenderAddress);
      runScript( "OnPINMsgAlert" );
      AlertHistory.Instance.AddAlert(XML);
      return true;
    }
  }
}
